cve_bin_tool/checkers/bind.py

```diff
@@ -12,7 +12,11 @@
 
 
 class BindChecker(Checker):
-    CONTAINS_PATTERNS = []
+    CONTAINS_PATTERNS = [
+        r"bind9_check_key",
+        r"bind9_check_namedconf",
+        # r"/bind9.xsl",
+    ]
     FILENAME_PATTERNS = [r"named"]
     VERSION_PATTERNS = [r"BIND ([0-9]+\.[0-9]+\.[0-9]+)"]
```

Comment:
While searching for strings in `bind`, there were no strings with `VERSION_PATTERNS = [r"BIND ([0-9]+\.[0-9]+\.[0-9]+)"]` 
in `.deb` packages. But, this string was present in `.rpm`, `.tgz`

The `CONTAINS_PATTERNS ` contains the strings, which were common in all packages (.rpm, .deb, .tgz).

Reply:
This is the output that I got for `.rpm` packages:

```
ol/helper_script.py bind-9.16.6-20.39.x86_64.rpm
isc_socket_bind
bind9_check_key
bind9_check_namedconf
cfg_type_bindkeys
ldap_bind_s
ldap_unbind_s
libbind9.so.1600
named_g_defaultbindkeys
/etc/bind.keys
BIND
named version: BIND 9.16.6 <25846cf>
BIND 9 is maintained by Internet Systems Consortium,
corporation.  Support and training for BIND 9 are
'--host=x86_64-suse-linux-gnu' '--build=x86_64-suse-linux-gnu' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/lib' '--localstatedir=/var' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--disable-dependency-tracking' '--with-python=/usr/bin/python3' '--includedir=/usr/include/bind' '--disable-static' '--with-openssl' '--enable-threads' '--with-libtool' '--with-libxml2' '--with-libjson' '--with-libidn2' '--with-dlz-mysql' '--with-dlz-ldap' '--with-randomdev=/dev/urandom' '--enable-ipv6' '--with-pic' '--disable-openssl-version-check' '--with-tuning=large' '--with-geoip' '--with-dlopen' '--with-gssapi=yes' '--disable-isc-spnego' '--enable-fixed-rrset' '--enable-filter-aaaa' '--with-systemd' '--enable-full-report' 'build_alias=x86_64-suse-linux-gnu' 'host_alias=x86_64-suse-linux-gnu' 'CFLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNO_VERSION_DATE' 'PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig'
server->bindkeysfile != ((void *)0)
_bind
version.bind/txt/ch disabled
bindkeys-file
binding control channel(s)
UDP/IPv4 socket bind failures
UDP4BindFail
UDP/IPv6 socket bind failures
UDP6BindFail
TCP/IPv4 socket bind failures
TCP4BindFail
TCP/IPv6 socket bind failures
TCP6BindFail
UnixBindFail
FDwatch socket bind failures
FdwatchBindFail
/bind9.xsl
type="text/xsl" href="/bind9.xsl"
Unix domain socket bind failures
LDAP driver failed while sending data to BIND.
LDAP driver could not bind connection number %u to server.
bindkeys-file "/etc/bind.keys";
try-tcp-refresh yes; /* BIND 8 compat */
#  Zones in the "_bind" view are NOT counted in the count of zones.
view "_bind" chaos {
zone "version.bind" chaos {
zone "hostname.bind" chaos {
zone "authors.bind" chaos {
#  sysconfdir/bind.keys doesn't exist).
# The bind.keys file is used to override the built-in DNSSEC trust anchors
# which are included as part of BIND 9.  The only trust anchors it contains
# file with a current version.  The latest version of bind.keys can always
# be obtained from ISC at https://www.isc.org/bind-keys.
<title>ISC BIND 9 Statistics</title>
<h1>ISC Bind 9 Configuration and Statistics</h1>
bind9_check_namedconf
libbind9.so.1600
VERSION_PATTERN = ['libbind9.so.1600', 'named version: BIND 9.16.6 <25846cf>', "'--host=x86_64-suse-linux-gnu' '--build=x86_64-suse-linux-gnu' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/lib' '--localstatedir=/var' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--disable-dependency-tracking' '--with-python=/usr/bin/python3' '--includedir=/usr/include/bind' '--disable-static' '--with-openssl' '--enable-threads' '--with-libtool' '--with-libxml2' '--with-libjson' '--with-libidn2' '--with-dlz-mysql' '--with-dlz-ldap' '--with-randomdev=/dev/urandom' '--enable-ipv6' '--with-pic' '--disable-openssl-version-check' '--with-tuning=large' '--with-geoip' '--with-dlopen' '--with-gssapi=yes' '--disable-isc-spnego' '--enable-fixed-rrset' '--enable-filter-aaaa' '--with-systemd' '--enable-full-report' 'build_alias=x86_64-suse-linux-gnu' 'host_alias=x86_64-suse-linux-gnu' 'CFLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNO_VERSION_DATE' 'PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig'", 'libbind9.so.1600']
FILENAME_PATTERN = ['named', 'named-checkconf']
VENDOR_PRODUCT = [('isc', 'bind')]
```

and this for `.deb` packages:

```
ol/helper_script.py bind9_9.16.15-1_amd64.deb
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
bind9_check_key
bind9_check_namedconf
libbind9-9.16.15-Debian.so
'--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=/usr/include' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-option-checking' '--disable-silent-rules' '--libdir=/usr/lib/x86_64-linux-gnu' '--runstatedir=/run' '--disable-maintainer-mode' '--disable-dependency-tracking' '--libdir=/usr/lib/x86_64-linux-gnu' '--sysconfdir=/etc/bind' '--with-python=python3' '--localstatedir=/' '--enable-threads' '--enable-largefile' '--with-libtool' '--enable-shared' '--enable-static' '--with-gost=no' '--with-openssl=/usr' '--with-gssapi=/usr' '--with-libidn2' '--with-json-c' '--with-lmdb=/usr' '--with-gnu-ld' '--with-maxminddb' '--with-atf=no' '--enable-ipv6' '--enable-rrl' '--enable-filter-aaaa' '--disable-native-pkcs11' '--enable-dnstap' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -ffile-prefix-map=/build/bind9-QWnomE/bind9-9.16.15=. -fstack-protector-strong -Wformat -Werror=format-security -fno-strict-aliasing -fno-delete-null-pointer-checks -DNO_VERSION_DATE -DDIG_SIGCHASE' 'LDFLAGS=-Wl,-z,relro -Wl,-z,now' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2'
/bind9.xsl
type="text/xsl" href="/bind9.xsl"
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
/usr/lib/debug/.dwz/x86_64-linux-gnu/bind9.debug
VERSION_PATTERN = ['libbind9-9.16.15-Debian.so', "'--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=/usr/include' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-option-checking' '--disable-silent-rules' '--libdir=/usr/lib/x86_64-linux-gnu' '--runstatedir=/run' '--disable-maintainer-mode' '--disable-dependency-tracking' '--libdir=/usr/lib/x86_64-linux-gnu' '--sysconfdir=/etc/bind' '--with-python=python3' '--localstatedir=/' '--enable-threads' '--enable-largefile' '--with-libtool' '--enable-shared' '--enable-static' '--with-gost=no' '--with-openssl=/usr' '--with-gssapi=/usr' '--with-libidn2' '--with-json-c' '--with-lmdb=/usr' '--with-gnu-ld' '--with-maxminddb' '--with-atf=no' '--enable-ipv6' '--enable-rrl' '--enable-filter-aaaa' '--disable-native-pkcs11' '--enable-dnstap' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -ffile-prefix-map=/build/bind9-QWnomE/bind9-9.16.15=. -fstack-protector-strong -Wformat -Werror=format-security -fno-strict-aliasing -fno-delete-null-pointer-checks -DNO_VERSION_DATE -DDIG_SIGCHASE' 'LDFLAGS=-Wl,-z,relro -Wl,-z,now' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2'"]
FILENAME_PATTERN = ['named']
VENDOR_PRODUCT = [('isc', 'bind')]
```

Reply:
we might need to add a debian version of `VERSION_STRING` too then.

Reply:
I've updated the `VERSION_PATTERNS`. Now it should be scanning for .deb versions too...
I made the regex based on `bind9-9.16.15`